apps/frontend/core-dapp/src/features/trade/close-trade/CloseTradeStore.ts

```diff
@@ -1,35 +1,81 @@
 import { BigNumber } from 'ethers'
-import { makeAutoObservable } from 'mobx'
+import { makeAutoObservable, reaction, runInAction } from 'mobx'
 import { validateStringToBN } from 'prepo-utils'
 import { RootStore } from '../../../stores/RootStore'
 import { MarketEntity } from '../../../stores/entities/MarketEntity'
 import { PositionEntity } from '../../../stores/entities/Position.entity'
+import { WEI_DENOMINATOR } from '../../../lib/constants'
 
 export class CloseTradeStore {
   input: { type: 'max' } | { type: 'input'; value: string } = {
     type: 'input',
     value: '',
   }
+  amountOutBN?: BigNumber = undefined
   constructor(private root: RootStore) {
     makeAutoObservable(this, {}, { autoBind: true })
+    this.subscribeClosePositionAmountOut()
+  }
+
+  private subscribeClosePositionAmountOut(): void {
+    reaction(
+      () => ({
+        action: this.root.tradeStore.action,
+        selectedPosition: this.selectedPosition,
+        amountBN: this.amountBN,
+        marketValuation: this.selectedMarket?.estimatedValuation,
+      }),
+      async ({ action, amountBN, selectedPosition }) => {
+        if (action !== 'close') return
+        const { address: collateralAddress } = this.root.collateralStore
+
+        if (
+          amountBN === undefined ||
+          !collateralAddress ||
+          !selectedPosition?.token.address ||
+          selectedPosition.pool.poolImmutables?.fee === undefined
+        )
+          return
+
+        const { fee } = selectedPosition.pool.poolImmutables
+
+        const output = await this.root.swapStore.quoteExactInput({
+          amountBN,
+          fromAddress: selectedPosition.token.address,
+          toAddress: collateralAddress,
+          fee,
+        })
+
+        runInAction(() => {
+          if (output === undefined) return
+          this.amountOutBN = output.output
+        })
+      }
+    )
+  }
+
+  // clean up when the tab is left
+  cleanUp(): void {
+    this.amountOutBN = undefined
+    this.setInput('')
```

Comment:
```suggestion
  // clear state when the tab is left
  clear(): void {
    this.amountOutBN = undefined
    this.setInput('')
```

Super small nit, but `cleanUp` makes me think that the `CloseTradeStore` is being destroyed and not used anymore. `clear` makes me think that its state is being reset. Also `reset` could be a good name.